packages/nextjs/components/SearchProjects.tsx

```diff
@@ -1,38 +1,77 @@
 import React, { useState } from "react";
 import Link from "next/link";
+import debounce from "lodash.debounce";
 import Autosuggest from "react-autosuggest";
+import useSWR from "swr";
 import { MagnifyingGlassIcon } from "@heroicons/react/24/outline";
+import { ListDocument } from "~~/models/List";
 import { ProjectDocument } from "~~/models/Project";
+import { fetcher } from "~~/utils/fetcher";
 
-const getSuggestions = (value: any, data: any) => {
-  const inputValue = value.trim().toLowerCase();
-  const inputLength = inputValue.length;
-
-  return inputLength === 0
-    ? []
-    : data.filter((item: any) => item.name.toLowerCase().slice(0, inputLength) === inputValue);
-};
-
-const getSuggestionValue = (suggestion: any) => suggestion.name;
-
-const renderSuggestion = (suggestion: any) => {
-  if (suggestion.isNoMatch) {
-    return <div className="text-gray-500">No matches found</div>;
-  }
-  return (
-    <Link href={`projects/${suggestion._id}`}>
-      <div>{suggestion.name}</div>
-    </Link>
-  );
-};
-
-interface Props {
-  data: ProjectDocument[];
+interface SearchResult {
+  searchData: ProjectDocument[] | ListDocument[];
 }
-const SearchProjects: React.FC<Props> = ({ data }) => {
+
+const SearchProjects: React.FC = () => {
   const [value, setValue] = useState("");
   const [suggestions, setSuggestions] = useState([]);
   const [isInputFocused, setInputFocused] = useState(false);
+  const [cachedSearch, setCachedSearch] = useState<any>({}); // local cache
+
+  const { data: initialData } = useSWR<SearchResult[]>(`/api/search?limit=50`, fetcher);
+
+  const fetchData = async (searchTerm: string) => {
+    if (!searchTerm.length) return [];
+    // Check the cache first
+    if (cachedSearch && cachedSearch[searchTerm]) {
+      return cachedSearch[searchTerm];
+    }
+    try {
+      const data = await fetch(`/api/search?term=${searchTerm}`);
+      const jsonData = await data.json();
+      setCachedSearch({ ...cachedSearch, [searchTerm]: jsonData });
+      return jsonData;
+    } catch (e) {
+      console.log("ERR_FETCHING_DATA:", e);
+    }
+  };
+  const renderSuggestion = (suggestion: any) => {
+    if (suggestion.isNoMatch && suggestion.isLoading) return <div className="text-gray-500">Searching...</div>;
+    if (suggestion.isNoMatch) {
+      return <div className="text-gray-500">No matches found</div>;
+    }
+    return (
+      <Link href={`${suggestion.type === "project" ? "projects" : "lists"}/${suggestion._id}`}>
```

Comment:
Not sure how to do this but if you are on a project page and you perform a search it sends you to a 404 because is tries to go to http://localhost:3001/projects/projects/64dca2d338a6eb4e8f87ad14. Just a small tweak I imagine to make not relative.

Reply:
Yeah, its fixed now.